feat(sumologic-image-widget) : Hubot can display widget from a sumologic dashboard

julien-vidal opened the pull request:
:construction: :construction: **_Work in progress_** :construction: :construction:

Commit:
Some test to get dashboard widget data

- Sumo API still in beta on that part, doesn't work well
on dashboard description and especially on widget description
the part we need of course ;)

Commit:
Start implement dashboard show cmd

- (test only) info available to get dashboard info
- (test only) data available to get dashboard data
- (wip) show, for now you have to enter a combo
dashboardid-widgetid for display a picture of a dash.
- exclude dump.rdb created by local hubot

julien-vidal committed:
feat(package) : Set package info and dev task

- Dev task just run hubot locally

julien-vidal committed:
feat(sumologic-image-widget) : Test gist, shit and doc it

firehist commented:
:rocket:

firehist commented on scripts/dashboard.coffee:
```diff
@@ -49,6 +63,7 @@ module.exports = (robot) ->
                 Dashboard URL manager
                 `!dashboard set <url>` : Switch to given URL
                 `!dashboard sumo <dashboardId>` : Switch to the dashboardId URL
+                `!dashboard show <dashboardId>-<widgetId>` : Get picture of a widget
```
In my mind, dashboard means the TV Dashboard :)
Maybe use a sumologic cmd ? :)

julien-vidal replied:
This one is just for testing, will disappear when finished ^^

firehist replied:
for sure, feel free it's a dashboard TV manager, I don't have TV

julien-vidal replied:
You should buy one :stuck_out_tongue_winking_eye: 
Really under construction lol, I think I will move my dev in an other place at the end

firehist replied:
For sure I should buy one.
Waiting our office keys :lollipop:

firehist commented on scripts/hubot-sumologic-image-widget/main.coffee:
```diff
@@ -0,0 +1,47 @@
+# Input - Widget data from sumologic
+# 1) Use combinaison of jsdom and highchart to generate an svg graph
+# 2) Use svg2img to generate image file
+# 3) Render it in an accessible place but temporary
```
Permanently through s3 storage ;)
Almost free :dancers:

julien-vidal replied:
Better, hubot can directly send the generated file, last step I didn't test :)

julien-vidal replied:
I confirm, no need to s3, locally hubot can send a generated file.
Last test on heroku...

firehist replied:
So cool :)
S3 was for public purpose (if you want share the link to other people we don't have access to slack)

julien-vidal committed:
feat(sumologic-image-widget) : Play with sumologic data and revertengine

- Find the logic between sumologic data and
how use them in a stacked bar chart.

Commit:
feat(sumologic-image-widget) : Push some refactor on dashboard...

julien-vidal committed:
feat(sumologic-image-widget) : Start implementing show widget

- Check readme im tired ;)

firehist commented on scripts/hubot-sumologic-image-widget/sendToSlack.coffee:
```diff
@@ -0,0 +1,21 @@
+
+  console.log "Ready to slack it !"
+
+  form.submit("https://slack.com/api/files.upload", (err, res) ->
```
So simple :)
You rock :guitar:

julien-vidal replied:
So nice ! But it's not finished yet !

Commit:
Merge branch 'payou-playground-graph-img-generation' of https://github.com/Boulangerie/hubot-mill into payou-playground-graph-img-generation

Commit:
feat(sumologic-image-widget) : Start stackedBarFormatter

- Format chart of type stackedBar (Waouuu)
- Clean and tidy mocks
- Next: Finish stackedBar format data for highchart

Commit:
feat(sumologic-image-widget) : First working version

- stackedBar works but it is ugly

julien-vidal committed:
feat(sumologic-image-widget) : Hubot listen for sumo command

julien-vidal committed:
feat(sumologic-image-widget) : Add Pie formatter

julien-vidal committed:
feat(sumologic-image-widget) : Add some others chart to implement

Commit:
feat(sumologic-image-widget) : Add missing mocks for front dashboard

- Bonus : Roadmap updated

julien-vidal committed:
feat(sumologic-image-widget) : remove sumo token from code

julien-vidal committed:
feat(sumologic-image-widget) : Refactor chain architecture

- Implement Promise and allow chainability
- Untie widgetFinder/GraphBuilder/sendToSlack
- sumo command is now composed of each logic step.
- Formatters disappear replaced by Charts
- Charts use a lib like highchart or not and each one is responsible to
how they will render in a dom window.
- Hubot can be teleported to be shared accross classes
- Some clean
- Highcharts helper should be done
- SvgHelper will be implemented soon
- (WIP) Non tested yet !

julien-vidal committed:
feat(sumologic-image-widget) : Refactor fixing

- Update readme todo ^^

julien-vidal committed:
feat(sumologic-image-widget) : add linesChart support

- Use highcharts on backstage

julien-vidal committed:
feat(sumologic-image-widget) : add (time)counter charts

- They are just template to fill for now
- Implemented in graphBuilder

julien-vidal committed:
feat(sumologic-image-widget) : Refactor how graphBuilder getGraph

- Previously we expect the constructor of a graph
class to return the jsDom document object. This wasn't a good idea
now each graph class have to implement a method getSvg which is
responsible to return the expected svg string.
- This allow us to do what we want in graph class instead of adding
a hidden dep to jsdom...

julien-vidal committed:
feat(sumologic-image-widget) : Start implementing (time)counter

- This graph doesn't rely on highcharts, they are too
simple ! We directly use svg template for them. They
are pretty simple for now and aren't able to color them
like sumo. We don't have this rules so we can't do that
for now.
- Next : Bind this graphs to sumo data !

julien-vidal committed:
feat(sumologic-image-widget) : (time)counter Done

julien-vidal committed:
feat(sumologic-image-widget) : Update roadmap

julien-vidal committed:
feat(sumologic-image-widget) : Fix how counter transform data

- Update roadmap too
- Transformation from sumo is too simple and doesn't handle
all cases.

Commit:
feat(sumologic-image-widget) : Add clean way to store graph file

- now : graphBuilder try to use temp file
- next : Test it / export code to a dedicated class

julien-vidal committed:
feat(sumologic-image-widget) : Implement tmp folder and clean

- Working code but completely ugly
- Refactor it in clean class in next commit plz !

Commit:
feat(sumologic-image-widget) : Start refactor of file management

Commit:
feat(sumologic-image-widget) : Implement fileHelper

- SendToslack left for integration.

Commit:
feat(sumologic-image-widget) : End FileHelper integration

Commit:
feat(sumologic-image-widget) : Add Highcharts theme support

- Now need a beautiful one ;)

julien-vidal committed:
feat(sumologic-image-widget) : Implement how to learn to hubot

- File uploaded with a special name can be considered
as a source of knowledge by hubot
- He try to get this file, parse it as json and extend his
memory.
- Brain is used for persistence.
- WidgetFinder is now based on it
- Readme updated

julien-vidal committed:
feat(sumologic-image-widget) : Comment dark theme

- not removed for let a footprint of how this
works

Commit:
feat(sumologic-image-widget) : Tidy file by purposes

Commit:
feat(sumologic-image-widget) : Remap dependencies due to previous tidy

Commit:
feat(sumologic-image-widget) : Add info message when hubot learn new tricks

Commit:
feat(sumologic-image-widget) : Rename method for erase sumo memory

-> lobotomize sumo

Commit:
feat(sumologic-image-widget) : Clean fileHelper

Commit:
feat(sumologic-image-widget) : Clean todo and clean

- SendToSlack should work everywhere
- We log a bit more the main process
- Todo removed and done :troll:

julien-vidal committed:
feat(sumologic-image-widget) : Add notification on hubot learnings

- Error to finish

Commit:
feat(sumologic-image-widget) : End error message on hubot learning fail
